nemo/collections/nlp/models/token_classification/token_classification_model.py

```diff
@@ -148,7 +148,10 @@ def validation_step(self, batch, batch_idx):
 
         preds = torch.argmax(logits, axis=-1)[subtokens_mask]
         labels = labels[subtokens_mask]
-        tp, fp, fn = self.classification_report(preds, labels)
+        self.classification_report(preds, labels)
```

Comment:
@ericharper, can you review the changes to this file? I mostly copied from text_classification_model.py.
Specifically, can you check that the instantiation of self.classification_report makes sense in this file.

Reply:
What's the difference between ['macro', 'micro', 'weighted']? Should everything just be micro?

Reply:
'micro' is accuracy, all these are different ways of aggregating tp, tn, fp for each label and calculating the final result. 'Macro' should be the default.

Reply:
ClassificationReport is used in 4 models:
IntentSlotClassificationModel
TextClassificationModel
PunctuationCapitalizationModel
TokenClassificationModel

Can I ask for a review on each model so we can approve of the changes?

Reply:
Updated all of the above models.